### PR TITLE
Image Studio: Enqueue assets in Jetpack plugins

### PR DESCRIPTION
**Pull request by kat3samsin**
Fixes #

## Proposed changes:

* Add enqueueing of Image Studio assets in Jetpack plugins to enable Image Studio in Simple, Atomic, and Self-hosted sites.

Checks for both Unified Chat flag or `jetpack_image_studio_enabled` filter

~~Out of scope: Self hosted sites.~~

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

**WPCOM Sandbox** 
1. Install to your sandbox:
```
bin/jetpack-downloader test jetpack forno-167/image-studio-jetpack-plugin
```
2. Deploy Big Sky PR 5597
3. Enable Image Studio by running in sandbox
```
echo "<?php add_filter( 'jetpack_image_studio_enabled', '__return_true' );" > ~/public_html/wp-content/mu-plugins/enable-image-studio.php
```

OR

Enabling Unified Chat, see pgatNH-1CI-p2

4. Go to Media Library 
Verify in Network tab `image-studio.min.js` is loaded from `widgets.wp.com` and `image-studio.js` (from BigSky) is not
image

**Atomic**
~~Pre-requisite: This WP Calypso [PR](https://github.com/Automattic/wp-calypso/pull/108684) needs to be deployed first~~ Deployed!
 
1. Generate your test site with Jetpack Beta [here](https://href.li/?https://jurassic.ninja/create?jetpack-beta&branches.jetpack=add/mystuff)
2. Connect Jetpack
3. Go Jetpack Beta plugin > Settings > Jetpack > Manage > search for a feature branch > forno 167 > activate
4. Enable jetpack image studio flag
```
echo "<?php add_filter( 'jetpack_image_studio_enabled', '__return_true' );" > ~/htdocs/wp-content/mu-plugins/enable-image-studio.php
```
5. Go to Media Library > Generate Image > Verify you can generate an image without issues, also try editing
6. Go to Block Editor > Edit / Generate with Image Studio should work

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+ *
+ * - Media Library: load if either filter is true (no query param needed).
+ * - Block editors (Post/Site Editor): load only with `enable_image_studio=1` query param.
+ * - Other screens: don't load.
+ *
+ * @return bool
+ */
+function should_load_on_current_screen() {
+	return is_media_library() || ( is_block_editor() && has_enable_image_studio_param() );
+}
```
[the remark this answers is not in the document]
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:76]: `is_image_studio_enabled()` is already checked by all callers before this function is reached. This only determines valid screen types.

File: projects/plugins/jetpack/tests/php/extensions/plugins/image-studio/Image_Studio_Test.php [hunk 2]
```diff
@@ -0,0 +1,1333 @@
...
+	}
+
+	/**
+	 * Reset Jetpack Gutenberg extension availability.
+	 */
+	private function reset_availability() {
+		$reflection = new ReflectionClass( 'Jetpack_Gutenberg' );
+		$property   = $reflection->getProperty( 'availability' );
+		@$property->setAccessible( true ); // @codingStandardsIgnoreLine — needed for PHP < 8.1, suppressed for PHP 8.5+ deprecation.
+		$property->setValue( null, array() );
```
[the remark this answers is not in the document]
Comment (kat3samsin) [reply projects/plugins/jetpack/tests/php/extensions/plugins/image-studio/Image_Studio_Test.php:88]: Tests that use jetpack_set_available_extensions call apply_filters directly and don't depend on the callback surviving across test methods. No test-order issue here.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+ * (under ABSPATH). This avoids an HTTP round-trip and works on sandboxes where
+ * outbound requests to widgets.wp.com may be blocked.
+ *
+ * @return array|false The decoded asset data, or false on failure.
+ */
+function get_asset_data() {
+	$cached = get_transient( ASSET_TRANSIENT );
+	if ( false !== $cached ) {
+		return $cached;
+	}
```
[the remark this answers is not in the document]
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:108]: Resolved in [fa5a745](https://github.com/Automattic/jetpack/pull/47063/commits/fa5a745a7e3c55fd1c233415c2f6693450db8f27)

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+ * On WordPress.com, the asset file may be accessible on the local filesystem
+ * (under ABSPATH). This avoids an HTTP round-trip and works on sandboxes where
+ * outbound requests to widgets.wp.com may be blocked.
+ *
+ * @return array|false The decoded asset data, or false on failure.
+ */
+function get_asset_data() {
+	$cached = get_transient( ASSET_TRANSIENT );
+	if ( false !== $cached ) {
+		return $cached;
```
[the remark this answers is not in the document]
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:107]: The transient is intentionally checked first for performance. When SCRIPT_DEBUG is toggled on, delete_transient  
  or a fresh page load after the transient expires handles this naturally. The edge case is too narrow to justify  
  adding a conditional wrapper around every cache read.

File: projects/plugins/jetpack/tests/php/extensions/plugins/image-studio/Image_Studio_Test.php [hunk 5]
```diff
@@ -0,0 +1,1351 @@
...
+
+	/**
+	 * AI image extensions that Image Studio replaces.
+	 */
+	private const AI_IMAGE_EXTENSIONS = array(
+		'ai-featured-image-generator',
+		'ai-assistant-image-extension',
+		'ai-general-purpose-image-generator',
+		'ai-assistant-experimental-image-generation-support',
+	);
```
Comment (ariskataoka) [inline projects/plugins/jetpack/tests/php/extensions/plugins/image-studio/Image_Studio_Test.php]:
```suggestion
	private const AI_IMAGE_EXTENSIONS =ImageStudio\get_ai_image_extensions();
```

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+		$data = get_asset_data_from_remote();
+	}
+
+	if ( false === $data ) {
+		return false;
+	}
+
+	if ( ! ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ) {
+		set_transient( ASSET_TRANSIENT, $data, HOUR_IN_SECONDS );
+	}
```
Comment (ariskataoka) [inline projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:121]: Nit: If an asset fetch fails, we’ll retry it on every page load (no negative caching), which could be noisy during a CDN outage.
That said, it looks consistent with the rest of the repo - I didn’t find any asset fetch doing negative caching today

**Review by ariskataoka — COMMENTED** [review]
THanks for working on this! Code looks good overall!
I'm curious about testing steps for a self-hosted environment? Or what to check in my sandbox once the filter is added.

**Comment by kat3samsin** [comment]
> THanks for working on this! Code looks good overall!
I'm curious about testing steps for a self-hosted environment? Or what to check in my sandbox once the filter is added.

Thank you for checking @ariskataoka! The expectation is that we shouldn't need to have BigSky plugin installed. And once this `jetpack_image_studio_enabled', '__return_true` is enabled it should also load Image Studio in Media Library and Post/Site editor (with args). Update the PR description I to include the correct steps! 

I think this PR won't work yet on self-hosted site until we move the loading of Agents Manager headless here (Jetpack plugin). I'll update the PR description to add this info. I'm still testing this in a self-hosted site using JN.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 6 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:121]: Yeah I'll leave this as is for now.

File: projects/plugins/jetpack/tests/php/extensions/plugins/image-studio/Image_Studio_Test.php (hunk 5 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/tests/php/extensions/plugins/image-studio/Image_Studio_Test.php]: Resolved in [683b3a7](https://github.com/Automattic/jetpack/pull/47063/commits/683b3a712bd39559631ae4024998ea32bac9390f)

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+	'jetpack_set_available_extensions',
+	function ( $extensions ) {
+		return array_merge(
+			(array) $extensions,
+			array(
+				FEATURE_NAME,
+			)
+		);
+	}
+);
```
Comment (t-wright) [inline projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php]: We may not need this. We already do `\Jetpack_Gutenberg::set_extension_available( FEATURE_NAME );` and have added it in `extensions/index.json`

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+/**
+ * Register the Image Studio plugin.
+ *
+ * Registers unconditionally when either filter is true. Screen-level gating
+ * happens at enqueue time since get_current_screen() is not available here.
+ *
+ * @return void
+ */
+function register_plugin() {
+	if ( ! is_image_studio_enabled() ) {
```
Comment (t-wright) [inline projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:87]:
```suggestion
	if ( ! is_image_studio_enabled() || ! has_enable_image_studio_param() ) {
```

This will ensure we are correctly setting extension availability in `window.Jetpack_Editor_Initial_State.available_blocks` which we can use to decide when to override the image gen entry points.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+		return;
+	}
+
+	// When the screen is available, only disable if Image Studio will actually load.
+	if ( function_exists( 'get_current_screen' ) && get_current_screen() ) {
+		if ( ! should_load_on_current_screen() ) {
+			return;
+		}
+	}
```
Comment (t-wright) [inline projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:287]:
```suggestion
	if ( ! \Jetpack_Gutenberg::is_available( FEATURE_NAME ) ) {
		return;
	}
```

I don't see this used anywhere else in the repo but seems made for this purpose and would simplify the conditional logic here. `should_load_on_current_screen()` would also no longer be required with this change

Comment (t-wright) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php]: `should_load_on_current_screen()` would no longer be required with this change

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 8 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:87]: I think registration happens before get_current_screen() is available, so it can't do screen-level gating here. Screen gating is handled at enqueue time by enqueue_image_studio() and enqueue_image_studio_admin().

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 9 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:287]: `is_available()` only checks global registration, not screen context. Tried to apply but it disables Jetpack AI in the Post Editor when Image Studio isn't enabled.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 7 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php]: Applied in [5f524ee](https://github.com/Automattic/jetpack/pull/47063/commits/5f524eeed3e078e3673a77b43f6d2e786b485db4)

File: projects/plugins/jetpack/changelog/forno-167-image-studio-jetpack-plugin [hunk 10]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: enhancement
+
+Image Studio: add Image Studio plugin support to block editor and media library.
```
[the remark this answers is not in the document]
Comment (kat3samsin) [reply projects/plugins/jetpack/changelog/forno-167-image-studio-jetpack-plugin:4]: There is a period.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 8 above)
Comment (t-wright) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:87]: Do we need `get_current_screen()` for this? I think `jetpack_register_gutenberg_extensions` runs off `enqueue_block_assets` so it will only run in block editor contexts?

The problem my comment was attempting to address is that currently `window.Jetpack_Editor_Initial_State.available_blocks` shows `image-studio` as available even without `enable_image_studio` query string. This seems to be the main mechanism for determining feature availability in Jetpack so I feel like it's important for it to be accurate.

Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:87]: > Do we need get_current_screen() for this? I think jetpack_register_gutenberg_extensions runs off enqueue_block_assets so it will only run in block editor contexts?

You’re right @t-wright that in this code path it runs in block editor contexts.
My thinking is we still need two distinct gates:

Capability gate: Image Studio is enabled globally via `jetpack_image_studio_enabled`.
Context gate: only load/activate on opted-in editor requests via enable_image_studio=1 (and screen checks as needed).
So Image Studio can be globally enabled by filter, but still intentionally unavailable on specific block-editor screens (for example Post Editor or Site Editor) unless the request is opted in.

**Comment by kat3samsin** [comment]
Converting this draft again. 

I'll remove `enable_image_studio=1` param.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 8 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:87]: I removed `enable_image_studio` param 😅

Comment (t-wright) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php:87]: Well it was an interesting conversation even if it was redundant 😅

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+ * Register the Image Studio headless agent provider with the agents manager.
+ *
+ * When jetpack_image_studio_enabled is true, adds the Image Studio headless
+ * agent provider module so the agents manager can load it.
+ *
+ * @param array $providers Existing agent provider module IDs.
+ * @return array Modified array of provider module IDs.
+ */
+function register_headless_agent_provider( $providers ) {
+	if ( ! apply_filters( 'jetpack_image_studio_enabled', false ) ) {
```
[the remark this answers is not in the document]
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php]: Resolved in [e532dea](https://github.com/Automattic/jetpack/pull/47063/commits/e532dea9af31679fa5c95c16a3aba22f6a478a09)

**Comment by kat3samsin** [comment]
@t-wright if you have time today, can you give this another look? I already removed the `enable_image_studio` check 😅 Thank you!

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php [hunk 12]
```diff
@@ -0,0 +1,333 @@
...
+	if ( ! is_image_studio_enabled() ) {
+		return;
+	}
+
+	// When the screen is available, only disable if Image Studio will actually load.
+	if ( function_exists( 'get_current_screen' ) && get_current_screen() ) {
+		if ( ! should_load_on_current_screen() ) {
+			return;
+		}
+	}
```
Comment (t-wright) [inline projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php]:
```suggestion
	if ( ! function_exists( 'get_current_screen' ) || ! get_current_screen() || ! should_load_on_current_screen() ) {
		return;
	}
```

If screen is not available, image studio won't load so i think we would want to return in that case too?

**Review by t-wright — DISMISSED** [review]
Thanks for working on this! It tested well for me. Left one comment which we may want to address. 

I tested on simple, atomic, and JT. I wasn't able to test generation via JT due to auth errors but image studio was enabled under the correct conditions.

File: projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php (hunk 12 above)
Comment (kat3samsin) [reply projects/plugins/jetpack/extensions/plugins/image-studio/image-studio.php]: Applied https://github.com/Automattic/jetpack/pull/47063/commits/0bc21b84d5e5f9f3e7dffc9a20eea3c3ff2d2de2

**Comment by kat3samsin** [comment]
Tested in Simple, Atomic, and Self Hosted sites.
